### PR TITLE
Accommodate optional attendees.

### PR DESCRIPTION
**Pull request by Sami-2000**
Make minor edits to the test cases so that they build. Create a new, broader query function which will accommodate optional attendees if possible.

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -21,9 +21,28 @@
 import java.util.Iterator;
 
 public final class FindMeetingQuery {
+  /* If we can accomodate optional attendees, do so. Otherwise if there are required attendees,
+  ignore optional attendees.*/
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    Collection<String> totalAttendees = new ArrayList<String>();
+    totalAttendees.addAll(request.getAttendees());
+    totalAttendees.addAll(request.getOptionalAttendees());
```
Comment (josephdhurst) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Nit: the method name `getAttendees` makes me think that it gets all attendees. From the context in here I can tell that `getAttendees` only gives required attendees, but you might use it in other places in the future and forget to add context. Also having to add context every time you call a method is tedious.

You may want to rename it to something like `getRequiredAttendees`.

Additionally, variable names that begin with total are usually numbers. I'd suggest renaming `totalAttendees` to `allAttendees`.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -21,9 +21,28 @@
...
+    }
+    else {
+      return optResults;
+    }
+  }
+
   /* Identify conflicting events, then iterating through the day, skipping conflicts, and 
   identifying time ranges longer than the duration of the requested meeting. */
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+  public Collection<TimeRange> query2(Collection<Event> events, MeetingRequest request) {
```
Comment (josephdhurst) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Adding a number like 2 to the end of a method doesn't give the reader any extra information (unless `2` has some meaning here that I'm not aware of). I think you should just keep this function's name `query`.

**Review by josephdhurst — APPROVED** [review]

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (Sami-2000) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: I will definitely change the totalAttendees variable! However, the getAttendees function was actually written for us (we did not create the meeting request class). I worry that changing its name may affect other pre-written dependencies.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 2 above)
Comment (Sami-2000) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: The tests (most of which we did not write) depend on our overall 'main' function being named 'query.' Therefore, I can name my internal query function anything except 'query' (unless I want two functions named query). I will try queryInternal, that at least gives more info than query2?

**Pull request merged**